multimodal/health_multimodal/image/model/model.py

```diff
@@ -0,0 +1,173 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+import enum
+from dataclasses import dataclass
+from typing import Any, Optional, Tuple, Union, Sequence
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from pl_bolts.models.self_supervised.resnets import resnet18, resnet50
+
+from .modules import MLP, MultiTaskModel
+
+TypeImageEncoder = Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]
+
+
+@enum.unique
+class ResnetType(str, enum.Enum):
+    RESNET18 = "resnet18"
+    RESNET50 = "resnet50"
+
+
+@dataclass
+class ImageModelOutput():
+    img_embedding: torch.Tensor
+    patch_embedding: torch.Tensor
+    projected_global_embedding: torch.Tensor
+    class_logits: torch.Tensor
+    projected_patch_embeddings: torch.Tensor
+
+
+class ImageModel(nn.Module):
+    """Image encoder module"""
+
+    def __init__(self,
+                 img_model_type: str,
+                 joint_feature_size: int,
+                 freeze_encoder: bool = False,
+                 pretrained_model_path: Optional[str] = None,
+                 **downstream_classifier_kwargs: Any):
```

Comment:
A reminder that the docstrings should explain what `downstream_classifier_kwargs` are used for.